crates/iota-core/src/execution_cache/cache_types.rs

```diff
@@ -293,10 +294,12 @@ where
             let mut entry = entry.value().lock();
             check_ticket()?;
 
-            // Ticket expiry makes this assert impossible.
-            // TODO: relax to debug_assert?
-            assert!(!entry.is_newer_than(&value), "entry is newer than value");
-            *entry = value;
+            // Ticket expiry should make this assert impossible.
+            if entry.is_newer_than(&value) {
+                debug_fatal!("entry is newer than value");
+            } else {
+                *entry = value;
```

Comment:
Now this happens when "entry is not newer than value". Should it be when "value is newer than entry"?

Reply:
Do you mean to change the logic into the other way around?

Reply:
I am just asking what should be the right logic. If adding objects with the same version is allowed (why? would that be a duplicate?), then it's fine as it is, otherwise I think we should change it.